pkg/operator/controller/ingress/controller.go

```diff
@@ -713,6 +720,56 @@ func (r *reconciler) ensureIngressDeleted(ingress *operatorv1.IngressController)
 	return utilerrors.NewAggregate(errs)
 }
 
+// GCS: findMostRecentIngress and findCondition copied from openshift-router:status.go...curious if there is a better way
+
+// findMostRecentIngress returns the name of the ingress status with the most recent Admitted condition transition time,
+// or an empty string if no such ingress exists.
+func (r *reconciler) clearRouteStatus(icName string) error {
+	routeList := &routev1.RouteList{}
+
+	// List all routes.
+
+	start := time.Now()
+	if err := r.client.List(context.TODO(), routeList); err != nil {
+		return fmt.Errorf("failed to list all routes in order to clear route status")
+	}
+	elapsed := time.Since(start)
+	fmt.Println(elapsed)
+
+	// Go through each route and clear status if admitted by this ingress controller.
+	start2 := time.Now()
+	for _, route := range routeList.Items {
+		updated := route.DeepCopy()
```

Comment:
The deepcopy could be moved to imediately before it's used below to avoid unnecessary copies.  

It might be worth declaring `var updated routev1.Route` here and then using `route.DeepCopyInto(&updated)` to do the deepcopy to save some memory allocations.

Reply:
> It might be worth declaring `var updated routev1.Route` here and then using `route.DeepCopyInto(&updated)` to do the deepcopy to save some memory allocations.

Actually, the variable declaration should be outside the outermost `for` loop to minimize allocations, if you decide to take this approach.

Reply:
ah of course, good idea